zilliqa/src/consensus.rs

```diff
@@ -302,27 +298,63 @@ impl Consensus {
     }
 
     fn download_blocks_up_to(&mut self, to: u64) -> Result<()> {
+
         for view in (self.view + 1)..to {
             self.view = view;
             self.block_store.request_block_by_view(view)?;
         }
-        self.view = to + 1;
+
```

Comment:
The prev logic here could allow you to effectively decrement your view

Reply:
Maybe `self.view = self.view.max(to + 1)` is simpler?